manual/manual/html_processing/src/process_api.ml

```diff
@@ -0,0 +1,414 @@
+    (* let reg_type = Str.regexp {|<code class="type">\(.+\)</code>|} in *)
+    let rec loop () = try
+        let line = input_line inch in
+        try let _ = Str.search_forward reg line 0 in
+          let code = parse line $ "code.type" in
+          let sig_txt = texts code
+                        |> String.concat ""
+                        |> String.escaped in
+
+          (* We now replace anonymous "t"'s by the qualified "Module.t" *)
+          let sig_txt = match mod_name with
+            | None -> sig_txt
+            | Some mod_name ->
+              Str.global_replace anon_t_regexp (mod_name ^ ".t") sig_txt in
+          (* let _ = Str.search_forward reg_type line 0 in
+           * let the_sig = Str.matched_group 1 line |> strip_a_tag in *)
+          sprintf "Signature=[%s]" sig_txt |> pr;
+          close_in inch; Some (to_string code |> String.escaped, sig_txt)
+        with
+        | Not_found -> loop ()
+      with
+      | End_of_file -> close_in inch; None in
+    loop ()
+
+  (* Example: "Buffer.html#VALadd_subbytes" ==> Some "VALadd_subbytes" *)
+  let get_id ref =
+    try let i = String.index ref '#' in
+      Some (String.sub ref (i+1) (String.length ref - i - 1))
+    with Not_found -> sprintf "Could not find id for %s" ref |> pr; None
+
+  let make ?(with_sig = true) () =
+    let ch = Scanning.open_in ("index_values.html"
+                               |> with_dir !src_dir) in
+    find ch "<table>";
+    let rec loop list =
+      if try find ch "<tr><td><a"; false with Not_found -> true
+      then list
+      else
+        (* Scan value reference *)
+        let val_ref = bscanf ch " href=%S" sid in
+        let val_name = bscanf ch ">%s@<" sid in
+        find ch  "[<a";
+
+        (* Scan module reference (ie. filename) *)
+        let mod_ref = bscanf ch " href=%S" sid in
+        let mod_name = bscanf ch ">%s@<" sid in
+        bscanf ch "/a>]</td> <td>" ();
+
+        let signature =
+          if with_sig then
+            get_id val_ref
+            |> flat_option (fun id -> get_sig ~mod_name ~id mod_ref)
+          else None in
+
+        (* Scan info *)
+        let info = match bscanf ch "<%s@>" sid with
+          | "/td" -> find ch "</tr>"; ""
+          | "div class=\"info\"" ->
+            let s = concat_before ch "</td></tr>" in
+            "<div class=\"info\">" ^ s
+          | s -> raise (Scan_failure s) in
+        pr val_ref;
+
+        let new_entry = ((mod_name, mod_ref), (val_name, val_ref), info, signature) in
+        loop (new_entry :: list) in
+    loop []
+    |> extract_infotext
+end
+(******************************************)
+
+let save_index file index =
+  let outch = open_out file in
+  output_string outch "var GENERAL_INDEX = [\n";
+  index
+  |> List.iter (fun (mdule, value, infolist, signature) ->
+      fprintf outch {|["%s", "%s", "%s", "%s", "%s", "%s", "%s", "%s"],|}
+        (fst mdule) (snd mdule) (fst value) (snd value) (fst infolist) (snd infolist)
```

Comment:
... will eliminate all the `fst` and `snd` calls here

Reply:
done